### PR TITLE
RDoc-1187 InsufficientMemoryException when running on encrypted db

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown [hunk 1]
```diff
@@ -231,4 +231,27 @@ Under construction
 
 ## Encryption Issues  
 
-Under construction
+### Insufficient Memory Exception
+
+With encrypted databases we lock some memory in order to avoid leaking secrets to disk. 
+
+This means that if we run out of memory, the OS is not allowed to page it to disk. By default RavenDB treats this as a catastrophic error and will not continue the operation.
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown]: Need to also explain how to increase the OS limit here.
Need to explain _why_ we are doing this

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown [hunk 2]
```diff
@@ -231,4 +231,27 @@ Under construction
...
+
+With encrypted databases we lock some memory in order to avoid leaking secrets to disk. 
+
+This means that if we run out of memory, the OS is not allowed to page it to disk. By default RavenDB treats this as a catastrophic error and will not continue the operation.
+
+You can change this behavior, see the [Security Configuration Section](../../server/configuration/security-configuration#security.donotconsidermemorylockfailureascatastrophicerror).
+
+{CODE-BLOCK:plain}
+Memory exception occurred: System.InsufficientMemoryException: Failed to increase the min working set size so we can lock 4,294,967,296 for D:\so-encrypt-test\RavenData\Databases\SO\Indexes\Auto_Questions_ByBody\Temp\compression.0000000000.buffers. With encrypted databases we lock some memory in order to avoid leaking secrets to disk. Treating this as a catastrophic error and aborting the current operation. ---> System.ComponentModel.Win32Exception: Insufficient system resources exist to complete the requested service
+   at System.Diagnostics.Process.SetWorkingSetLimitsCore(Nullable`1 newMin, Nullable`1 newMax, IntPtr& resultingMin, IntPtr& resultingMax)
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown]: Do we really need entire the entire stacktrace here?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown [hunk 3]
```diff
@@ -231,4 +231,27 @@ Under construction
...
+### Insufficient Memory Exception
+
+With encrypted databases we lock some memory in order to avoid leaking secrets to disk. 
+
+This means that if we run out of memory, the OS is not allowed to page it to disk. By default RavenDB treats this as a catastrophic error and will not continue the operation.
+
+You can change this behavior, see the [Security Configuration Section](../../server/configuration/security-configuration#security.donotconsidermemorylockfailureascatastrophicerror).
+
+{CODE-BLOCK:plain}
+Memory exception occurred: System.InsufficientMemoryException: Failed to increase the min working set size so we can lock 4,294,967,296 for D:\so-encrypt-test\RavenData\Databases\SO\Indexes\Auto_Questions_ByBody\Temp\compression.0000000000.buffers. With encrypted databases we lock some memory in order to avoid leaking secrets to disk. Treating this as a catastrophic error and aborting the current operation. ---> System.ComponentModel.Win32Exception: Insufficient system resources exist to complete the requested service
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown]: Let's change `D:\so-encrypt-test` (**test** part doesn't look good in the docs) to `D:\RavenDB` for example

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown [hunk 4]
```diff
@@ -231,4 +231,27 @@ Under construction
 
 ## Encryption Issues  
 
-Under construction
+### Insufficient Memory Exception
+
+With encrypted databases we lock some memory in order to avoid leaking secrets to disk.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown]: WRT "some memory" - we can be a bit more explicit here and write that we lock the memory of temporary buffers that are memory mapped files (e.g. `compression.0000000000.buffers`)

**Comment by iftahbe** [comment]
Closing, need to rewrite this

**Pull request closed without merging**